src/scripts/OSFramework/OSUI/Pattern/OverflowMenu/OverflowMenu.ts

```diff
@@ -0,0 +1,326 @@
+			}
+
+			Helper.A11Y.AriaExpanded(this.selfElement, this.isOpen.toString());
+		}
+
+		/**
+		 * Set the callbacks
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.OverflowMenu.OverflowMenu
+		 */
+		protected setCallbacks(): void {
+			this._eventBalloonOnToggle = this._balloonOnToggleCallback.bind(this);
+			this._eventOnClick = this._onClickCallback.bind(this);
+		}
+
+		/**
+		 * Method to set the event listeners
+		 *
+		 * @protected
+		 * @memberof OverflowMenu
+		 */
+		protected setEventListeners(): void {
+			this._triggerElem.addEventListener(GlobalEnum.HTMLEvent.Click, this._eventOnClick);
+
+			Event.DOMEvents.Listeners.GlobalListenerManager.Instance.addHandler(
+				Event.DOMEvents.Listeners.Type.BalloonOnToggle,
+				this._eventBalloonOnToggle
+			);
+		}
+
+		/**
+		 * Update info based on htmlContent
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.OverflowMenu.OverflowMenu
+		 */
+		protected setHtmlElements(): void {
+			this._triggerElem = Helper.Dom.ClassSelector(this.selfElement, Enum.CssClass.Trigger);
+			this._balloonElem = Helper.Dom.ClassSelector(this.selfElement, Enum.CssClass.Balloon);
+		}
+
+		/**
+		 * Method to unset the callbacks.
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.OverflowMenu.OverflowMenu
+		 */
+		protected unsetCallbacks(): void {
+			this._eventBalloonOnToggle = undefined;
+			this._eventOnClick = undefined;
+		}
+
+		/**
+		 * Removes the local value of the variables pointing to HTML elements;
+		 *
+		 * @protected
+		 * @memberof OSFramework.Patterns.OverflowMenu.OverflowMenu
+		 */
+		protected unsetHtmlElements(): void {
+			this._balloonElem = undefined;
+			this._triggerElem = undefined;
+			this._balloonFeature = undefined;
+		}
+
+		/**
+		 * Method to build the Pattern
+		 *
+		 * @memberof OverflowMenu
+		 */
+		public build(): void {
+			super.build();
+			this.setHtmlElements();
+			this.setA11YProperties();
+			this._setBalloonFeature();
+			this._setOverflowMenuShape();
+			this.setCallbacks();
+			this.setEventListeners();
+			this.finishBuild();
+		}
+
+		/**
+		 * Method to change the value of configs/current state.
+		 *
+		 * @param {string} propertyName
+		 * @param {unknown} propertyValue
+		 * @memberof OverflowMenu
+		 */
+		public changeProperty(propertyName: string, propertyValue: unknown): void {
+			super.changeProperty(propertyName, propertyValue);
+
+			if (this.isBuilt) {
+				switch (propertyName) {
+					case Enum.Properties.Position:
+						this._balloonFeature.updatePositionOption(propertyValue as GlobalEnum.FloatingPosition);
+						break;
+					case Enum.Properties.Shape:
+						this._setOverflowMenuShape(propertyValue as GlobalEnum.ShapeTypes);
+						this._balloonFeature.setBalloonShape(propertyValue as GlobalEnum.ShapeTypes);
+						break;
+				}
+			}
+		}
+
+		/**
+		 * Method to close the Pattern
+		 *
+		 * @memberof OverflowMenu
+		 */
+		public close(): void {
```

Comment:
Shouldn't we have the same "flag" about if it was done by the API? in order to avoid the same as we're avoiding at the open method?

Reply:
No need, as the bodyClickCallback already has a validation for  if (this.isOpen), so it never reaches here.